src/openshift/component.ts

```diff
@@ -163,6 +163,15 @@ export class Component extends OpenShiftItem {
         return null;
     }
 
+    private static async validateComponentName(value: string, application: OpenShiftObject) {
+        const componentList: Array<OpenShiftObject> = await Component.odo.getComponents(application);
+        return componentList.map((component) => {
```

Comment:
This is not how you should do this 👎 try to use `Array.indexOf()` or `Array.includes()`.

Reply:
I think we can't use indexof() and includes() because we don't have the reference of the object here. We only have the component name as string which is a property of the object inside array. So we need to use a higher order functions like `Array.prototype.find()` or `Array.prototype.findIndex()`.